Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1754,6 +1754,29 @@ bool RenderBox::getBackgroundPaintedExtent(const LayoutPoint& paintOffset, Layou
     return !geometry.hasNonLocalGeometry;
 }
 
+LayoutRect RenderBox::backgroundClipRect() const
+{
+    // FIXME: Check for the maximum background clip rect.
+    switch (style().backgroundClip()) {
+    case FillBox::BorderBox:
+        // A 'border-box' clip on scrollable elements local attachment is treated as 'padding-box'.
+        // https://www.w3.org/TR/css-backgrounds-3/#the-background-attachment
+        if (!style().isOverflowVisible() && style().backgroundLayers().attachment() == FillAttachment::LocalBackground)
```

Comment:
This code checks `style().backgroundLayers().attachment()`, so the return value for this function is only valid for the first background. This adds ambiguity. The correct way to write this function would be to pass in the `FillAttachment`, and then the caller has to say which background layer they are asking the question for.

Reply:
The `style.backgroundClip()` check has the same issue (I think that helper probably shouldn't exist).

This code change isn't changing `BackgroundPainter::paintFillLayer`, which I think is needed.